pkg/reconciler/autoscaling/kpa/scaler.go

```diff
@@ -370,3 +373,20 @@ func (ks *scaler) scale(ctx context.Context, pa *autoscalingv1alpha1.PodAutoscal
 	logger.Infof("Scaling from %d to %d", currentScale, desiredScale)
 	return desiredScale, ks.applyScale(ctx, pa, desiredScale, ps)
 }
+func (ks *scaler) revisionTimeout(ctx context.Context, pa *autoscalingv1alpha1.PodAutoscaler) (int64, error) {
+	gvr := schema.GroupVersionResource{
+		Group:    "serving.knative.dev",
+		Version:  "v1",
+		Resource: "revisions",
+	}
+	uRevision, err := ks.dynamicClient.Resource(gvr).Namespace(pa.Namespace).Get(ctx, pa.Name, metav1.GetOptions{})
```

Comment:
good point.
in case this proposal is viable, where is a good place in the code to fetch the `revisionTimeout`?

just to note that this is called only when we are scaling to 0.

Reply:
> In case this proposal is viable, where is a good place in the code to fetch the revisionTimeout?

Two options
- we add a field to the Knative PodAutoscaler type
- we propagate this using an annotation on the PodAutoscaler type

Reply:
Looks like we already have an annotation 

https://github.com/knative/serving/blob/f939498a917c252e6da9057ff9e51a08d1c9e185/pkg/apis/autoscaling/v1alpha1/pa_lifecycle.go#L171-L174

Reply:
Oddly seems like it's a user knob that we propagate to the deployment - but we aren't setting it on all PodAutoscalers (eg. when the user doesn't set it)

https://github.com/knative/serving/blob/f939498a917c252e6da9057ff9e51a08d1c9e185/pkg/apis/serving/register.go#L146

Reply:
I created a new annotation `serving.knative.dev/revision-timeout` for the pod autoscaler, this has the RevisionTimeout field on the RevisionSpec. With this change I am not fetching the revision from the API server